### PR TITLE
FIX: change all integer types to longin/longout, enum to mbbi/mbbo

### LINKED ISSUE #47: Integer tc_types should map to RTYP longin/longout, not ai/ao

### LINKED ISSUE #68: Beckhoff Enum should use mbbi/mbbo
https://github.com/slaclab/pytmc/blob/3b0b3e83f631cb3030b2709244be3332766f346b/pytmc/xml_collector.py#L604

### PR DESCRIPTION
**Pull request by klauer**
Closes #47 
Closes #68  
Related to #66 

`asynUint32Digital` support has been removed from this PR for now, as it's not supported by the ESS's adsAsynPortDriver. While it's not difficult to add this support (I have a branch that's 50% of the way there), we need to weigh up the cost of either getting it merged back or maintaining our own fork. Re-addition of this unsigned code may be revisited later after further discussions.

### CODE REVIEW

File: docs/pragma_usage.rst [hunk 1]
```diff
@@ -15,7 +15,9 @@ TwinCAT data types and their corresponding record types are as follows:
 +-----------+--------------------------+-------------------------+--------------+-----------------+--------------------+-----------------------------------------+
 | Data type |       Lower bound        |       Upper bound       | Memory space |   Record Type   |    Scalar DTYP     |              Waveform DTYP              |
 +===========+==========================+=========================+==============+=================+====================+=========================================+
-| BYTE      | 0                        | 255                     | 8 bit        | bi, bo          |  asynUInt32Digital | asynInt8ArrayIn, asynInt8ArrayOut       |
+| BOOL      | 0                        | 1                       | 8 bit        | bi, bo          |  asynUInt32Digital | asynInt8ArrayIn, asynInt8ArrayOut       |
```
Comment (teddyrendahl) [inline docs/pragma_usage.rst]: These tables are no fun to create...

Comment (klauer) [reply docs/pragma_usage.rst]: https://ozh.github.io/ascii-tables/

Comment (teddyrendahl) [reply docs/pragma_usage.rst]: Bookmarked

**Comment by klauer** [comment]
OK, I think this is ready for review.

File: pytmc/xml_collector.py [hunk 2]
```diff
@@ -678,6 +674,24 @@ def __str__(self):
 }
 
 
+tc_type_num_bits = {
```
Comment (teddyrendahl) [inline pytmc/xml_collector.py]: Almost seems like this should be in `pyads` but maybe it is easier to keep our own map.

File: tests/test_xml_collector.py [hunk 3]
```diff
@@ -948,13 +981,16 @@ def test_BaseRecordPackage_guess_PREC(example_singular_tmc_chains,
...
+                                      tc_type, io, is_str, is_arr, final_FTVL,
+                                      dbd_file):
     record = BaseRecordPackage(851, example_singular_tmc_chains[0])
     record.chain.last.tc_type = tc_type
     record.chain.last.is_array = is_arr
     record.chain.last.is_str = is_str
     record.generate_naive_config()
     record.cfg.add_config_line('io', io, overwrite=True)
+    result = record.guess_type()
+    result = record.guess_DTYP()
```
Comment (teddyrendahl) [inline tests/test_xml_collector.py]: No assertions for `DTYP`?

File: tests/test_xml_collector.py [hunk 4]
```diff
@@ -650,21 +652,21 @@ def test_BaseRecordPackage_guess_TSE():
     ("BOOL", "i", True, 'waveform'),
     ("BOOL", "o", True, 'waveform'),
     ("BOOL", "io", True, 'waveform'),
-    ("INT", "i", False, 'ai'),
-    ("INT", "o", False, 'ao'),
-    ("INT", "io", False, 'ao'),
+    ("INT", "i", False, 'longin'),
```
Comment (teddyrendahl) [inline tests/test_xml_collector.py]: 🎉

File: tests/test_xml_collector.py (hunk 3 above)
Comment (klauer) [reply tests/test_xml_collector.py]: Added

File: pytmc/xml_collector.py (hunk 2 above)
Comment (klauer) [reply pytmc/xml_collector.py]: Yeah. These dictionaries could certainly live elsewhere (even in this repo, `xml_collector` is not ideal). 

Also, `tc_type_num_bits` is actually unused after the removal of `asynUInt32Digital` mask support. I can remove it if desirable.

Comment (teddyrendahl) [reply pytmc/xml_collector.py]: If we aren't using it, remove it and keep it clean imo

Comment (klauer) [reply pytmc/xml_collector.py]: Removed

**Comment by teddyrendahl** [comment]
@klauer do you want to wait and test this against a running PLC?

**Comment by klauer** [comment]
@teddyrendahl I think this could be merged and tested/fixed later. I don't expect many issues as this is not a large departure from the current implementation.

File: pytmc/xml_obj.py [hunk 5]
```diff
@@ -333,6 +333,11 @@ def add_config_field(self, f_name, f_set, line_no=None, config=None,
         None
 
         """
+        f_set = str(f_set)
+
+        if not f_set.startswith('"') or not f_set.endswith('"'):
```
Comment (n-wbrown) [inline pytmc/xml_obj.py]: Adds some much needed consistency. 👍

**Pull request merged**